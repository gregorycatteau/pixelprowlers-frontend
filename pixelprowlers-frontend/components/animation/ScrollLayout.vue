<template>
  <section class="scroll-section">
    <Hero :scrollFactor="scrollFactor" :isActive="currentSection === 'hero'" />
    <Problems :scrollFactor="scrollFactor" :isActive="currentSection === 'problems'" />
    <WhoIAm :scrollFactor="scrollFactor" :isActive="currentSection === 'whoiam'" />

    <div v-if="props.debugVisible" class="debug">
      <p>üéØ SF: {{ scrollFactor.toFixed(2) }}</p>
      <p>üìç Section: {{ currentSection }}</p>
    </div>
  </section>
</template>


<script setup lang="ts">
import { ref, watch, computed, onMounted, onBeforeUnmount, provide } from 'vue'
import { useScrollLock } from '@vueuse/core'
import gsap from 'gsap'
import Hero from '@/components/home/hero.vue'
import Problems from '@/components/home/problems.vue'
import WhoIAm from '@/components/home/whoIAm.vue'

const props = withDefaults(defineProps<{
  debugVisible?: boolean
}>(), {
  debugVisible: false
})

const scrollFactor = ref(0)
const scrollUnlocked = ref(true)
const maxScrollFactor = 3

const currentSection = computed(() => {
  const sf = scrollFactor.value
  if (sf < 1) return 'hero'
  if (sf >= 1 && sf < 1.5) return 'problems'
  if (sf >= 1.5 && sf < 2.2) return 'whoiam'
  return 'next'
})

provide('scrollFactor', scrollFactor)

onMounted(() => {
  const scrollLock = useScrollLock(document.body)
  scrollLock.value = true

  watch(scrollUnlocked, (v) => {
    scrollLock.value = !v
  })

  window.addEventListener('wheel', onWheel, { passive: true })
  console.log('üì¶ scrollLayout mounted ‚Äì scrollFactor is ready to rock')
})

onBeforeUnmount(() => {
  window.removeEventListener('wheel', onWheel)
})

function onWheel(e: WheelEvent) {
  if (!scrollUnlocked.value) {
    console.warn("‚õî Scroll is locked")
    return
  }

  const delta = e.deltaY
  const nextValue = scrollFactor.value + delta / 800
  scrollFactor.value = Math.min(Math.max(nextValue, 0), maxScrollFactor)
}

watch(scrollFactor, (val) => {
  if (val > 0.01 && !scrollUnlocked.value) {
    scrollUnlocked.value = true
    console.log('üîì Scroll unlocked via mouvement')
  }
})
</script>


<style scoped>
@reference "@/assets/css/main.css";

.scroll-section {
  @apply fixed top-0 left-0 w-full h-screen overflow-hidden;
  z-index: 0;
  background: transparent;
}

.debug {
  @apply fixed top-2 left-2 text-sm bg-black/80 text-green-400 px-4 py-2 rounded z-50;
  font-family: monospace;
  pointer-events: none;
}
</style>
