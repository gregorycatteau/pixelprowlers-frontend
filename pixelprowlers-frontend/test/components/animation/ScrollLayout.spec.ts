// 📁 test/components/ScrollLayout.spec.ts

import { mount } from '@vue/test-utils'
import ScrollLayout from '@/components/animation/ScrollLayout.vue'
import Hero from '@/components/home/hero.vue'
import Problems from '@/components/home/problems.vue'
import WhoIAm from '@/components/home/whoIAm.vue'
import { describe, it, expect, vi, beforeEach } from 'vitest'

describe('ScrollLayout.vue', () => {
  let wrapper: ReturnType<typeof mount>

  beforeEach(() => {
    wrapper = mount(ScrollLayout, {
      props: {
        debugVisible: true
      }
    })
  })

  it('se monte correctement', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('rend les sections enfants Hero, Problems et WhoIAm', () => {
    expect(wrapper.findComponent(Hero).exists()).toBe(true)
    expect(wrapper.findComponent(Problems).exists()).toBe(true)
    expect(wrapper.findComponent(WhoIAm).exists()).toBe(true)
  })

  it('affiche la section debug si debugVisible est true', () => {
    const debug = wrapper.find('.debug')
    expect(debug.exists()).toBe(true)
    expect(debug.text()).toContain('SF')
  })

  it('cache la section debug si debugVisible est false', async () => {
    await wrapper.setProps({ debugVisible: false })
    expect(wrapper.find('.debug').exists()).toBe(false)
  })

  it('calcule correctement la section active selon scrollFactor', async () => {
    wrapper.vm.scrollFactor = 0.5
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.currentSection).toBe('hero')

    wrapper.vm.scrollFactor = 1.2
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.currentSection).toBe('problems')

    wrapper.vm.scrollFactor = 1.8
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.currentSection).toBe('whoiam')

    wrapper.vm.scrollFactor = 2.5
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.currentSection).toBe('next')
  })

  it('gère les valeurs extrêmes de scrollFactor', async () => {
    wrapper.vm.scrollFactor = -1
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.currentSection).toBe('hero')

    wrapper.vm.scrollFactor = 99
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.currentSection).toBe('next')
  })

  it('bloque le scroll si scrollUnlocked est false', async () => {
    wrapper.vm.scrollUnlocked = false
    const initialFactor = wrapper.vm.scrollFactor

    // Simuler appel direct
    if (typeof wrapper.vm.onWheel === 'function') {
      wrapper.vm.onWheel({ deltaY: 100 })
    } else {
      const event = new Event('wheel')
      window.dispatchEvent(event)
    }

    expect(wrapper.vm.scrollFactor).toBe(initialFactor)
  })

  it('snapshot global du HTML', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
