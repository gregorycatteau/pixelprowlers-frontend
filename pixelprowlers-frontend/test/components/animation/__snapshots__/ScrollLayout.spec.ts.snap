// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ScrollLayout.vue > snapshot global du HTML 1`] = `
"<section data-v-3916768a="" class="scroll-section">
  <div data-v-ec7dfd49="" data-v-3916768a="" class="hero-wrapper">
    <div class="hero-panels" data-v-ec7dfd49="">
      <div class="panel panel-left" data-v-ec7dfd49=""></div>
      <div class="panel panel-center" data-v-ec7dfd49=""></div>
      <div class="panel panel-right" data-v-ec7dfd49=""></div>
      <div class="cta" data-v-ec7dfd49=""><button class="cta-button" data-v-ec7dfd49="">Parlez-nous de vous</button></div>
    </div>
  </div>
  <section data-v-d8ac1fab="" data-v-3916768a="" class="problems-container">
    <div data-v-d8ac1fab="" class="parallax-background"></div><canvas data-v-5a7eb9f7="" data-v-d8ac1fab="" class="rain-canvas" width="1024" height="768"></canvas><!-- Lignes d’intro -->
    <div data-v-d8ac1fab="" class="intro-wrapper">
      <p data-v-d8ac1fab="" class="intro-line">Un froid silence entoure vos actions.</p>
      <p data-v-d8ac1fab="" class="intro-line">Personne ne voit. Personne n’entend.</p>
      <p data-v-d8ac1fab="" class="intro-line">Vous avancez dans la brume, seul, figé entre urgence et épuisement.</p>
      <p data-v-d8ac1fab="" class="intro-line">Et les outils censés aider ? Aussi glacés qu’inaccessibles.</p>
    </div><!-- Cartes -->
    <div data-v-d8ac1fab="" class="cards-wrapper">
      <div data-v-d8ac1fab="" class="problem-card">
        <div data-v-d8ac1fab="" class="card-icon"><img data-v-d8ac1fab="" src="/invisible.png" alt="" class="card-img"></div>
        <h3 data-v-d8ac1fab="" class="card-title">Impact gelé</h3>
        <p data-v-d8ac1fab="" class="card-description">Ce que vous bâtissez est réel, mais enseveli sous la neige de l’indifférence.</p>
      </div>
      <div data-v-d8ac1fab="" class="problem-card">
        <div data-v-d8ac1fab="" class="card-icon"><img data-v-d8ac1fab="" src="/time-missing.png" alt="" class="card-img"></div>
        <h3 data-v-d8ac1fab="" class="card-title">Temps figé</h3>
        <p data-v-d8ac1fab="" class="card-description">Chaque instant se fige dans le givre des urgences. Le sens se perd dans le froid.</p>
      </div>
      <div data-v-d8ac1fab="" class="problem-card">
        <div data-v-d8ac1fab="" class="card-icon"><img data-v-d8ac1fab="" src="/complexity.png" alt="" class="card-img"></div>
        <h3 data-v-d8ac1fab="" class="card-title">Technologie inhospitalière</h3>
        <p data-v-d8ac1fab="" class="card-description">Des interfaces glaciales, conçues sans vous. Un labyrinthe gelé où rien ne pousse.</p>
      </div>
    </div><!-- CTA -->
    <div data-v-d8ac1fab="" class="cta-container"><button data-v-d8ac1fab="" class="cta-button"> Brisez la glace. Reprenez le contrôle. </button></div>
  </section>
  <section data-v-f0f882fa="" data-v-3916768a="" class="whoiam-container">
    <!-- Texte d'intro par lettre -->
    <div data-v-f0f882fa="" class="intro-wrapper">
      <div data-v-f0f882fa="" class="intro-line"><span data-v-f0f882fa="" class="intro-char">J</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">i</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">p</span><span data-v-f0f882fa="" class="intro-char">a</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">a</span><span data-v-f0f882fa="" class="intro-char">g</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">c</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">.</span></div>
      <div data-v-f0f882fa="" class="intro-line"><span data-v-f0f882fa="" class="intro-char">J</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">i</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">a</span><span data-v-f0f882fa="" class="intro-char">r</span><span data-v-f0f882fa="" class="intro-char">t</span><span data-v-f0f882fa="" class="intro-char">i</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">a</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">d</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">w</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">b</span><span data-v-f0f882fa="" class="intro-char">.</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">U</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">d</span><span data-v-f0f882fa="" class="intro-char">é</span><span data-v-f0f882fa="" class="intro-char">v</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">l</span><span data-v-f0f882fa="" class="intro-char">o</span><span data-v-f0f882fa="" class="intro-char">p</span><span data-v-f0f882fa="" class="intro-char">p</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">r</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">q</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">i</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">a</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">l</span><span data-v-f0f882fa="" class="intro-char">o</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">g</span><span data-v-f0f882fa="" class="intro-char">t</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">m</span><span data-v-f0f882fa="" class="intro-char">p</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">é</span><span data-v-f0f882fa="" class="intro-char">t</span><span data-v-f0f882fa="" class="intro-char">é</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">d</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">l</span><span data-v-f0f882fa="" class="intro-char">'</span><span data-v-f0f882fa="" class="intro-char">a</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">t</span><span data-v-f0f882fa="" class="intro-char">r</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">c</span><span data-v-f0f882fa="" class="intro-char">ô</span><span data-v-f0f882fa="" class="intro-char">t</span><span data-v-f0f882fa="" class="intro-char">é</span><span data-v-f0f882fa="" class="intro-char">.</span></div>
      <div data-v-f0f882fa="" class="intro-line"><span data-v-f0f882fa="" class="intro-char">J</span><span data-v-f0f882fa="" class="intro-char">’</span><span data-v-f0f882fa="" class="intro-char">a</span><span data-v-f0f882fa="" class="intro-char">i</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">m</span><span data-v-f0f882fa="" class="intro-char">o</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">t</span><span data-v-f0f882fa="" class="intro-char">é</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">d</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">a</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">o</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">,</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">v</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">d</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">r</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">l</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">m</span><span data-v-f0f882fa="" class="intro-char">a</span><span data-v-f0f882fa="" class="intro-char">r</span><span data-v-f0f882fa="" class="intro-char">c</span><span data-v-f0f882fa="" class="intro-char">h</span><span data-v-f0f882fa="" class="intro-char">é</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">,</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">o</span><span data-v-f0f882fa="" class="intro-char">i</span><span data-v-f0f882fa="" class="intro-char">g</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">é</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">d</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">g</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">,</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">f</span><span data-v-f0f882fa="" class="intro-char">o</span><span data-v-f0f882fa="" class="intro-char">r</span><span data-v-f0f882fa="" class="intro-char">m</span><span data-v-f0f882fa="" class="intro-char">é</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">d</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">g</span><span data-v-f0f882fa="" class="intro-char">r</span><span data-v-f0f882fa="" class="intro-char">o</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">p</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">.</span></div>
      <div data-v-f0f882fa="" class="intro-line"><span data-v-f0f882fa="" class="intro-char">J</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">a</span><span data-v-f0f882fa="" class="intro-char">i</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">c</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">q</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">c</span><span data-v-f0f882fa="" class="intro-char">’</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">t</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">d</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">v</span><span data-v-f0f882fa="" class="intro-char">o</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">l</span><span data-v-f0f882fa="" class="intro-char">o</span><span data-v-f0f882fa="" class="intro-char">i</span><span data-v-f0f882fa="" class="intro-char">r</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">c</span><span data-v-f0f882fa="" class="intro-char">h</span><span data-v-f0f882fa="" class="intro-char">a</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">g</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">r</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">l</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">c</span><span data-v-f0f882fa="" class="intro-char">h</span><span data-v-f0f882fa="" class="intro-char">o</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">,</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">a</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">l</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">t</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">m</span><span data-v-f0f882fa="" class="intro-char">p</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">i</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">l</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">m</span><span data-v-f0f882fa="" class="intro-char">o</span><span data-v-f0f882fa="" class="intro-char">y</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">.</span></div>
      <div data-v-f0f882fa="" class="intro-line"><span data-v-f0f882fa="" class="intro-char">A</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">j</span><span data-v-f0f882fa="" class="intro-char">o</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">r</span><span data-v-f0f882fa="" class="intro-char">d</span><span data-v-f0f882fa="" class="intro-char">’</span><span data-v-f0f882fa="" class="intro-char">h</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">i</span><span data-v-f0f882fa="" class="intro-char">,</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">j</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">m</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">t</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">m</span><span data-v-f0f882fa="" class="intro-char">o</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">a</span><span data-v-f0f882fa="" class="intro-char">v</span><span data-v-f0f882fa="" class="intro-char">o</span><span data-v-f0f882fa="" class="intro-char">i</span><span data-v-f0f882fa="" class="intro-char">r</span><span data-v-f0f882fa="" class="intro-char">-</span><span data-v-f0f882fa="" class="intro-char">f</span><span data-v-f0f882fa="" class="intro-char">a</span><span data-v-f0f882fa="" class="intro-char">i</span><span data-v-f0f882fa="" class="intro-char">r</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">a</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">r</span><span data-v-f0f882fa="" class="intro-char">v</span><span data-v-f0f882fa="" class="intro-char">i</span><span data-v-f0f882fa="" class="intro-char">c</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">d</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">c</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">x</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">q</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">i</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">b</span><span data-v-f0f882fa="" class="intro-char">â</span><span data-v-f0f882fa="" class="intro-char">t</span><span data-v-f0f882fa="" class="intro-char">i</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">s</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">t</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">d</span><span data-v-f0f882fa="" class="intro-char">u</span><span data-v-f0f882fa="" class="intro-char">&nbsp;</span><span data-v-f0f882fa="" class="intro-char">l</span><span data-v-f0f882fa="" class="intro-char">i</span><span data-v-f0f882fa="" class="intro-char">e</span><span data-v-f0f882fa="" class="intro-char">n</span><span data-v-f0f882fa="" class="intro-char">.</span></div>
    </div><!-- Cartes -->
    <div data-v-f0f882fa="" class="cards-wrapper">
      <div data-v-f0f882fa="" class="whoiam-card">
        <div data-v-f0f882fa="" class="card-icon"><img data-v-f0f882fa="" src="/companion.jpg" alt="" class="card-img"></div>
        <h3 data-v-f0f882fa="" class="card-title">Un compagnon de route</h3>
        <p data-v-f0f882fa="" class="card-description">PixelProwlers, c’est une main tendue, pas une prestation froide. Une vraie aventure humaine.</p>
      </div>
      <div data-v-f0f882fa="" class="whoiam-card">
        <div data-v-f0f882fa="" class="card-icon"><img data-v-f0f882fa="" src="/solidarity.png" alt="" class="card-img"></div>
        <h3 data-v-f0f882fa="" class="card-title">Une offre solidaire</h3>
        <p data-v-f0f882fa="" class="card-description">Tarifs éthiques, solutions évolutives, accompagnement patient. Ici, on avance ensemble.</p>
      </div>
      <div data-v-f0f882fa="" class="whoiam-card">
        <div data-v-f0f882fa="" class="card-icon"><img data-v-f0f882fa="" src="/roots.png" alt="" class="card-img"></div>
        <h3 data-v-f0f882fa="" class="card-title">Des racines communes</h3>
        <p data-v-f0f882fa="" class="card-description">Je viens de là où vous êtes : terrain, terrain, terrain. Ce que je propose, je l’ai éprouvé.</p>
      </div>
    </div><!-- Citation finale -->
    <div data-v-f0f882fa="" class="quote-wrapper">
      <blockquote data-v-f0f882fa="" class="quote-text"> « Je code pour aimer, pour croire, pour bâtir. » </blockquote>
    </div>
  </section>
  <div data-v-3916768a="" class="debug">
    <p data-v-3916768a="">🎯 SF: 0.00</p>
    <p data-v-3916768a="">📍 Section: hero</p>
  </div>
</section>"
`;
